File: leetcode/java/can-place-flowers.java
class Solution {
    public boolean canPlaceFlowers(int[] flowerbed, int n){
        if(n==0){
            return true;
        }
        int count = 0;
        for (int i = 0; i < flowerbed.length; i++) {
           if(flowerbed[i]==0) {
            boolean isLeftZero = (i == 0 || flowerbed[i-1]==0);
            boolean isRightZero = (i==flowerbed.length-1 || flowerbed[i+1]==0);
            if(isLeftZero && isRightZero){
                flowerbed[i]=1;
                count++;
            }
           }
        }
        return count >= n;
    }
}

File: leetcode/python/can-place-flowers.py
from typing import List
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        flower_length = len(flowerbed)
        if flower_length == 0:
            return True
        #start from 0
        count = 0
        temp = [flowerbed[i] for i in range(n)]
        for i in range(0,flower_length):
            if i == 0 or temp[i-1] == 0:
                count+=1
                flowerbed

